### PR TITLE
Fix sonar blocker issues in PortableContextImpl

### LINKED ISSUE #19702: Fix sonar blocker issues in PortableContextImpl
Close `BufferObjectDataInput` in `lookupClassDefinition` method

https://sonarcloud.io/project/issues?fileUuids=AXIKzIDqbU5mxILfmQ42&id=hz-os-master&resolved=false&severities=BLOCKER&types=BUG

### PR DESCRIPTION
**Pull request by girijasameera**
<PR description here>

Fixes #19702(point out issues this PR fixes, if any)

Forward-port (backport) of: #NNNN (link to PR which is the basis for this PR, if applicable)

EE PR: #NNNN (link to enterprise counterpart PR)

Breaking changes (list specific methods/types/messages):
* API
* client protocol format
* serialized form
* snapshot format

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Add `Add to Release Notes` label if changes should be mentioned in release notes or `Not Release Notes content` if changes are not relevant for release notes
- [x] Request reviewers if possible
- [ ] New public APIs have `@Nonnull/@Nullable` annotations
- [ ] New public APIs have `@since` tags in Javadoc
- [ ] Send backports/forwardports if fix needs to be applied to past/future releases

### CODE REVIEW

**Comment by girijasameera** [comment]
@ldziedziul Please review

File: hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/portable/PortableContextImpl.java [hunk 1]
```diff
@@ -74,15 +74,16 @@ public ClassDefinition lookupClassDefinition(Data data) throws IOException {
...
-
-        BufferObjectDataInput in = serializationService.createObjectDataInput(data);
-        int factoryId = in.readInt();
-        int classId = in.readInt();
-        int version = in.readInt();
-
-        ClassDefinition classDefinition = lookupClassDefinition(factoryId, classId, version);
-        if (classDefinition == null) {
-            classDefinition = readClassDefinition(in, factoryId, classId, version);
+        ClassDefinition classDefinition = null;
```
Comment (ldziedziul) [inline hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/portable/PortableContextImpl.java:77]: What do you think about returning `classDefinition` in try-with-resources? This way you could limit the scope of the `classDefinition` variable

**Comment by ldziedziul** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/portable/PortableContextImpl.java (hunk 1 above)
Comment (girijasameera) [reply hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/portable/PortableContextImpl.java]: Thanks for the suggestion, I've made the change. Please take a look.

File: hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/portable/PortableContextImpl.java [hunk 2]
```diff
@@ -74,18 +74,18 @@ public ClassDefinition lookupClassDefinition(Data data) throws IOException {
...
 
-            classDefinition = lookupClassDefinition(factoryId, classId, version);
+            ClassDefinition classDefinition = lookupClassDefinition(factoryId, classId, version);
             if (classDefinition == null) {
                 classDefinition = readClassDefinition(in, factoryId, classId, version);
             }
+            return classDefinition;
         }
-        return classDefinition;
+        return null;
```
Comment (yemreinci) [inline hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/portable/PortableContextImpl.java:88]: You can also remove the return statement as it's unreachable.
```suggestion
```

File: hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/portable/PortableContextImpl.java [hunk 3]
```diff
@@ -74,17 +74,18 @@ public ClassDefinition lookupClassDefinition(Data data) throws IOException {
...
+            int version = in.readInt();
+
+            ClassDefinition classDefinition = lookupClassDefinition(factoryId, classId, version);
+            if (classDefinition == null) {
+                classDefinition = readClassDefinition(in, factoryId, classId, version);
+            }
+            return classDefinition;
         }
-        return classDefinition;
+        return null;
```
Comment (yemreinci) [inline hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/portable/PortableContextImpl.java:88]: You can also remove the return statement as it's unreachable.
```suggestion
```

**Comment by ldziedziul** [comment]
run-lab-run

**Comment by ldziedziul** [comment]
run-lab-run

**Review by ldziedziul — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**